File: src/002/p002.exs
defmodule Problem002 do
  defp fibonacci_sequence do
    Stream.unfold({1, 1}, fn {a, b} -> {a, {b, a + b}} end)
  end

  def solve do
    fibonacci_sequence()
    |> Stream.take_while(&(&1 < 4000000))
    |> Stream.filter(&(rem(&1, 2) == 0))
    |> Enum.sum()
  end
end

IO.puts Problem002.solve

File: src/002/p002.lua
function sum_even_fibs_below(limit)
  local sum = 0
  local a = 1
  local b = 2

  while b < limit do
    if b % 2 == 0 then
      sum = sum + b
    end
    a, b = b, a + b
  end
  return sum
end

function solve()
  return sum_even_fibs_below(4000000)
end

local result = solve()
print(result)
